### ISSUE TITLE
Script to run regional simulations similar to run_neon script

### ISSUE THREAD

**Issue by wwieder** [ESCOMP/CTSM#1533]
We want to provide a script to run regional simulations that's similar to the run_neon script. It will be built to use the inputdata files that results from the subset_data script. One important point about the regional case from single point cases is that mesh files are required, so that will change the setup a bit. Also single point cases set PTS_LAT/PTS_LON which won't be done for the regional case.

A separate but related task to this is to have a script to run a single points site after subset_data is run. There's likely a lot of infrastructure that the run_neon, run_tower, and run_regional scripts could have in common. We should try to structure them so they can share as much code as possible, which will make them much easier to maintain.

**_This is the original start of this issue from Will. We decided this overlaps with #1513 so we are separating it out to just this one issue..._**

regional simulations with nuopc will need to generate mesh files. @mvertens @ekluzek noted there are .ncl tools available for this, but @danicalombardozzi and @negin513 suggested this be wrapped into a `run_regional` wrapper script that does this for users.  

This functionally isn't really part of the NEON project, but it could leverage some of the python wrapper tools used in `run_neon` that have already been created by @jedwards4b and @negin513.

**Comment by ekluzek** [comment]
@wwieder this is pretty much a duplicate of #1513. The mesh file creation wouldn't go into run_regional it would go into subset_data. The thing I'm not clear on is exactly how we do that. One process would be to create a mesh file from the domain file that subset_data creates. That might be a mid-term solution, because in the end we don't need domain files. I'm just not sure about how to create a mesh file from subset_data.

In principle subset_data can also subset SCRIP grid files, but the problem is that they are in 1D vector format, so the subsetting doesn't look the same as the 2D grid files. I suppose it is the same for unstructured grids though.

**Comment by wwieder** [comment]
I'm happy to migrate this conversation over to #1513 to avoid duplicating issues.  Your call @ekluzek and @danicalombardozzi

**Comment by ekluzek** [comment]
Ahh, OK, thanks for the discussion. @danicalombardozzi I think what you are saying is that you want a run_regional script that works similar to the run_neon script. So that's a script that you run after subset_data to run a regional case. subset_data should create all the input datafiles needed to run the regional case, but you could use run_regional to create the case for you.

If that's correct, then I propose that we change this issue into one that's about making the run_regional script, and leave #1513 as it is.

**Comment by danicalombardozzi** [comment]
@ekluzek, yes, a 'run_regional' script would set up the case similar to the run_neon script. It would set up all the newly created datm and mesh files (and anything else required) in a case that a user could run. This would majorly simplify the process for running a regional simulation. We've also been discussing a similar script for single point simulations, similar to the run_neon, but for a user specified location rather than the NEON data we already have available.

**Comment by ekluzek** [comment]
We have some disscussion in #1565 that overlaps with this.

**Comment by ekluzek** [comment]
Polly B. and Sanjiv K. from the community are willing to be guinea pigs for this work on regional scripts.

**Comment by ekluzek** [comment]
I removed this as a requirement for removing MCT.  This would still be a good thing to do, but isn't required before removing support for MCT, as long as documentation is sufficient.

**Comment by wwieder** [comment]
It seems like we don't need to have this labeled as next, so I'm removing this label.

**Comment by ekluzek** [comment]
I thought this would be part of the hackathon we plan. So I started assigning the team to work on this. But, I realized this is about regional cases so I unassigned the team from it.

This still somewhat applies though in that we should be thinking about the python modules being able to work for both single-point and regional cases. There might be a separate tool (or python module like in subset_data), but we do want to have some part of it shared between single-point and regional cases.

**Comment by wwieder** [comment]
At this point I don't suggest we create and maintain a python wrapper to run regional cases.  Instead, clearer documentation that walks a user through subsetting data capabilities, mesh creation, and that provides an example of how to create a new case seems more appropriate?

**Comment by negin513** [comment]
@wwieder : This gist walks through the exact steps you outlined here for running regional cases: 
https://gist.github.com/negin513/d71268419f808b91a82cde5530d390b1

**Comment by samsrabin** [comment]
Closing as too hard and not useful.
